### PR TITLE
doc: Update docs on boothooks

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
doc: Update docs on boothooks

Improve explaination on #cloud-boothook for end-users.

Reorder and improve doc/rtd/explanation/format.rst.

SC-1657
Fixes GH-4542

Co-authored-by: Calvin Mwadime <calvin.mwadime@canonical.com>
```

## Additional Context

https://github.com/canonical/cloud-init/pull/5541#pullrequestreview-2196889423

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: doc/rtd/explanation/format.rst [hunk 1]
```diff
@@ -157,17 +157,50 @@ a MIME archive.
...
+:ref:`Network boot stage<boot-Network>`, even before ``cc_bootcmd``.
 
-Begins with: ``#cloud-boothook`` or ``Content-Type: text/cloud-boothook`` when
-using a MIME archive.
+This can be used when something has to be configured very early on boot,
+potentially on every boot, with less convenience as ``cc_bootcmd`` but more
+flexibility.
+
+.. note::
+   Boothooks are execute on every boot.
```
Comment (s-makin) [inline doc/rtd/explanation/format.rst:170]:
```suggestion
   Boothooks are executed on every boot.
```

**Review by s-makin — APPROVED** [review]
One tiny nit, otherwise LGTM :)

**Comment by aciba90** [comment]
Overlaps with: https://github.com/canonical/cloud-init/issues/4739

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Pull request merged**